### ISSUE TITLE
Cannot use ansible_managed in copy module with content option

### ISSUE THREAD

**Issue by sedlund** [ansible/ansible#34693]
##### ISSUE TYPE

 - Documentation Report
##### ANSIBLE VERSION

```
ansible 2.4.2.0
  config file = /home/ansible/ansible.cfg
  configured module search path = [u'/home/ansible/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.14 (default, Sep 23 2017, 22:06:14) [GCC 7.2.0]
```

##### SUMMARY

the copy module with the content option should be able to parse {{ ansible_managed }}

##### STEPS TO REPRODUCE

```
cat << EOF > test_copy_content.yml
- hosts: localhost
  gather_facts: no
  tasks:
    - name: test ansible_managed with copy module using content
      copy:
        dest: /tmp/test_copy.tmp
        content: |
          {{ ansible_managed | comment }}
          var: test
EOF
ansible-playbook -c local -l localhost test_copy_content.yml
```

##### EXPECTED RESULTS

Expected templating to work with the copy option as stated in documentation:

http://docs.ansible.com/ansible/latest/intro_configuration.html#ansible-managed
'ansible_managed
The ansible_managed string can be inserted into files written by Ansible’s config templating system:'

http://docs.ansible.com/ansible/latest/template_module.html
'You can use the copy module with the content: option if you prefer the template inline, as part of the playbook.'

##### ACTUAL RESULTS
Fatal Error

```
# ansible-playbook -c local -l localhost test_copy_content.yml   

PLAY [localhost] **************************************************************************************************************************************************

TASK [test ansible_managed with copy module using content] ********************************************************************************************************
fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'ansible_managed' is undefined\n\nThe error appears to have been in '/home/sedlund/test/test_copy_content.yml': line 4, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n  tasks:\n    - name: test ansible_managed with copy module using content\n      ^ here\n\nexception type: <class 'ansible.errors.AnsibleUndefinedVariable'>\nexception: 'ansible_managed' is undefined"}
        to retry, use: --limit @/home/sedlund/test/test_copy_content.retry

PLAY RECAP ********************************************************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   
```

**Comment by sivel** [comment]
I believe this to be a docs report rather than a bug.  This has never been supported outside the context of the `template` module.  In order to use `ansible_managed` it can only be accessed using `template`.

**Comment by paulfantom** [comment]
@sivel could it be a feature request? Adding this would simplify creating some files, especially ones using `to_nice_yaml` filter.

**Comment by Dmitriusan** [comment]
I'm also facing this issue. Why not just make the `ansible_managed` variable available for inline templates?

**Comment by devster31** [comment]
Also encountering this in `2.7.6`
```
ansible 2.7.6
  config file = /Users/devster/code/infrastructure/ansible/ansible.cfg
  configured module search path = ['/Users/devster/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/devster/code/infrastructure/ansible/env/lib/python3.7/site-packages/ansible
  executable location = /Users/devster/code/infrastructure/ansible/env/bin/ansible
  python version = 3.7.2 (default, Jan 13 2019, 12:51:54) [Clang 9.0.0 (clang-900.0.39.2)]
```
I do believe this should be added as a feature but if not possible, at least documented.

**Comment by bcoca** [comment]
Dupe of #37219 fixed in #39099
